frontend/src/components/GDataTableHeader.vue

```diff
@@ -0,0 +1,138 @@
+<!--
+SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+<template>
+  <tr>
+    <template
+      v-for="column in columns"
+      :key="column.key"
+    >
+      <th>
+        <div
+          class="v-data-table-header__content align-center table-header"
+          :class="[column.align ? `justify-${column.align}` : '', 'hover-button-container']"
+        >
+          <span
+            class="cursor-pointer"
+            @click="() => toggleSort(column)"
+          >{{ column.title }}</span>
+          <v-tooltip
+            v-if="column.sortable"
+          >
+            <template #activator="{ props: tooltipProps }">
```

Comment:
I think the problem here is that you cannot use parent activator in case of button, you would have to wrap the button in a div if I recall this properly. Need to double-check...